### PR TITLE
MGMT-4823 Remove spammy logs from the service

### PR DESCRIPTION
**Pull request by michaellevy101**
Removing the free-addresses long output from the post step reply step
The message was: time="2021-03-30T16:33:41Z" level=info msg="Received step reply <free-network-addresses-76210f63> from cluster <7af19149-96ab-4fbe-b5d5-00d5330dff2d> host <396bb210-6665-4254-a044-58fd657656fb>  exit-code <0> stdout <[{\"free_addresses\":[\"192.168.124.0\",\"192.168.124.2\",\"192.168.124.3\",\"192.168.124.4\",\"192.168.124.5\",\"192.168.124.6\",\"192.168.124.7\",\"192.168.124.8\",\"192.168.124.9\",\"192.168.124.15\",\"192.168.124.16\",\"192.168.124.17\",\"192.168.124.18\",\"192.168.124.19\",\"192.168.124.20\",\"192.168.124.21\",\"192.168.124.22\",\"192.168.124.23\",\"192.168.124.24\",\"192.168.124.25\",\"192.168.124.26\",\"192.168.124.27\",\"192.168.124.28\",\"192.168.124.29\",\"192.168.124.30\",\"192.168.124.31\",\"192.168.124.32\",\"192.168.124.33\",\"192.168.124.34\",\"192.168.124.35\",\"192.168.124.36\",\"192.168.124.37\",\"192.168.124.38\",\"192.168.124.39\",\"192.168.124.40\",\"192.168.124.41\",\"192.168.124.42\",\"192.168.124.43\",\"192.168.124.44\",\"192.168.124.45\",\"192.168.124.46\",\"192.168.124.47\",\"192.168.124.48\",\"192.168.124.49\",\"192.168.124.50\",\"192.168.124.51\",\"192.168.124.52\",\"192.168.124.53\",\"192.168.124.54\",\"192.168.124.55\",\"192.168.124.56\",\"192.168.124.57\",\"192.168.124.58\",\"192.168.124.59\",\"192.168.124.60\",\"192.168.124.61\",\"192.168.124.62\",\"192.168.124.63\",\"192.168.124.64\",\"192.168.124.65\",\"192.168.124.66\",\"192.168.124.67\",\"192.168.124.68\",\"192.168.124.69\",\"192.168.124.70\",\"192.168.124.71\",\"192.168.124.72\",\"192.168.124.73\",\"192.168.124.74\",\"192.168.124.75\",\"192.168.124.76\",\"192.168.124.77\",\"192.168.124.78\",\"192.168.124.79\",\"192.168.124.80\",\"192.168.124.81\",\"192.168.124.82\",\"192.168.124.83\",\"192.168.124.84\",\"192.168.124.85\",\"192.168.124.86\",\"192.168.124.87\",\"192.168.124.88\",\"192.168.124.89\",\"192.168.124.90\",\"192.168.124.91\",\"192.168.124.92\",\"192.168.124.93\",\"192.168.124.94\",\"192.168.124.95\",\"192.168.124.96\",\"192.168.124.97\",\"192.168.124.98\",\"192.168.124.99\",\"192.168.124.100\",\"192.168.124.101\",\"192.168.124.102\",\"192.168.124.103\",\"192.168.124.104\",\"192.168.124.105\",\"192.168.124.106\",\"192.168.124.107\",\"192.168.124.108\",\"192.168.124.109\",\"192.168.124.110\",\"192.168.124.111\",\"192.168.124.112\",\"192.168.124.113\",\"192.168.124.114\",\"192.168.124.115\",\"192.168.124.116\",\"192.168.124.117\",\"192.168.124.118\",\"192.168.124.119\",\"192.168.124.120\",\"192.168.124.121\",\"192.168.124.122\",\"192.168.124.123\",\"192.168.124.124\",\"192.168.124.125\",\"192.168.124.126\",\"192.168.124.127\",\"192.168.124.128\",\"192.168.124.129\",\"192.168.124.130\",\"192.168.124.131\",\"192.168.124.132\",\"192.168.124.133\",\"192.168.124.134\",\"192.168.124.135\",\"192.168.124.136\",\"192.168.124.137\",\"192.168.124.138\",\"192.168.124.139\",\"192.168.124.140\",\"192.168.124.141\",\"192.168.124.142\",\"192.168.124.143\",\"192.168.124.144\",\"192.168.124.145\",\"192.168.124.146\",\"192.168.124.147\",\"192.168.124.148\",\"192.168.124.149\",\"192.168.124.150\",\"192.168.124.151\",\"192.168.124.152\",\"192.168.124.153\",\"192.168.124.154\",\"192.168.124.155\",\"192.168.124.156\",\"192.168.124.157\",\"192.168.124.158\",\"192.168.124.159\",\"192.168.124.160\",\"192.168.124.161\",\"192.168.124.162\",\"192.168.124.163\",\"192.168.124.164\",\"192.168.124.165\",\"192.168.124.166\",\"192.168.124.167\",\"192.168.124.168\",\"192.168.124.169\",\"192.168.124.170\",\"192.168.124.171\",\"192.168.124.172\",\"192.168.124.173\",\"192.168.124.174\",\"192.168.124.175\",\"192.168.124.176\",\"192.168.124.177\",\"192.168.124.178\",\"192.168.124.179\",\"192.168.124.180\",\"192.168.124.181\",\"192.168.124.182\",\"192.168.124.183\",\"192.168.124.184\",\"192.168.124.185\",\"192.168.124.186\",\"192.168.124.187\",\"192.168.124.188\",\"192.168.124.189\",\"192.168.124.190\",\"192.168.124.191\",\"192.168.124.192\",\"192.168.124.193\",\"192.168.124.194\",\"192.168.124.195\",\"192.168.124.196\",\"192.168.124.197\",\"192.168.124.198\",\"192.168.124.199\",\"192.168.124.200\",\"192.168.124.201\",\"192.168.124.202\",\"192.168.124.203\",\"192.168.124.204\",\"192.168.124.205\",\"192.168.124.206\",\"192.168.124.207\",\"192.168.124.208\",\"192.168.124.209\",\"192.168.124.210\",\"192.168.124.211\",\"192.168.124.212\",\"192.168.124.213\",\"192.168.124.214\",\"192.168.124.215\",\"192.168.124.216\",\"192.168.124.217\",\"192.168.124.218\",\"192.168.124.219\",\"192.168.124.220\",\"192.168.124.221\",\"192.168.124.222\",\"192.168.124.223\",\"192.168.124.224\",\"192.168.124.225\",\"192.168.124.226\",\"192.168.124.227\",\"192.168.124.228\",\"192.168.124.229\",\"192.168.124.230\",\"192.168.124.231\",\"192.168.124.232\",\"192.168.124.233\",\"192.168.124.234\",\"192.168.124.235\",\"192.168.124.236\",\"192.168.124.237\",\"192.168.124.238\",\"192.168.124.239\",\"192.168.124.240\",\"192.168.124.241\",\"192.168.124.242\",\"192.168.124.243\",\"192.168.124.244\",\"192.168.124.245\",\"192.168.124.246\",\"192.168.124.247\",\"192.168.124.248\",\"192.168.124.249\",\"192.168.124.250\",\"192.168.124.251\",\"192.168.124.252\",\"192.168.124.253\",\"192.168.124.254\",\"192.168.124.255\"],\"network\":\"192.168.124.0/24\"},{\"free_addresses\":[\"192.168.142.0\",\"192.168.142.2\",\"192.168.142.3\",\"192.168.142.4\",\"192.168.142.5\",\"192.168.142.6\",\"192.168.142.7\",\"192.168.142.8\",\"192.168.142.9\",\"192.168.142.15\",\"192.168.142.16\",\"192.168.142.17\",\"192.168.142.18\",\"192.168.142.19\",\"192.168.142.20\",\"192.168.142.21\",\"192.168.142.22\",\"192.168.142.23\",\"192.168.142.24\",\"192.168.142.25\",\"192.168.142.26\",\"192.168.142.27\",\"192.168.142.28\",\"192.168.142.29\",\"192.168.142.30\",\"192.168.142.31\",\"192.168.142.32\",\"192.168.142.33\",\"192.168.142.34\",\"192.168.142.35\",\"192.168.142.36\",\"192.168.142.37\",\"192.168.142.38\",\"192.168.142.39\",\"192.168.142.40\",\"192.168.142.41\",\"192.168.142.42\",\"192.168.142.43\",\"192.168.142.44\",\"192.168.142.45\",\"192.168.142.46\",\"192.168.142.47\",\"192.168.142.48\",\"192.168.142.49\",\"192.168.142.50\",\"192.168.142.51\",\"192.168.142.52\",\"192.168.142.53\",\"192.168.142.54\",\"192.168.142.55\",\"192.168.142.56\",\"192.168.142.57\",\"192.168.142.58\",\"192.168.142.59\",\"192.168.142.60\",\"192.168.142.61\",\"192.168.142.62\",\"192.168.142.63\",\"192.168.142.64\",\"192.168.142.65\",\"192.168.142.66\",\"192.168.142.67\",\"192.168.142.68\",\"192.168.142.69\",\"192.168.142.70\",\"192.168.142.71\",\"192.168.142.72\",\"192.168.142.73\",\"192.168.142.74\",\"192.168.142.75\",\"192.168.142.76\",\"192.168.142.77\",\"192.168.142.78\",\"192.168.142.79\",\"192.168.142.80\",\"192.168.142.81\",\"192.168.142.82\",\"192.168.142.83\",\"192.168.142.84\",\"192.168.142.85\",\"192.168.142.86\",\"192.168.142.87\",\"192.168.142.88\",\"192.168.142.89\",\"192.168.142.90\",\"192.168.142.91\",\"192.168.142.92\",\"192.168.142.93\",\"192.168.142.94\",\"192.168.142.95\",\"192.168.142.96\",\"192.168.142.97\",\"192.168.142.98\",\"192.168.142.99\",\"192.168.142.100\",\"192.168.142.101\",\"192.168.142.102\",\"192.168.142.103\",\"192.168.142.104\",\"192.168.142.105\",\"192.168.142.106\",\"192.168.142.107\",\"192.168.142.108\",\"192.168.142.109\",\"192.168.142.110\",\"192.168.142.111\",\"192.168.142.112\",\"192.168.142.113\",\"192.168.142.114\",\"192.168.142.115\",\"192.168.142.116\",\"192.168.142.117\",\"192.168.142.118\",\"192.168.142.119\",\"192.168.142.120\",\"192.168.142.121\",\"192.168.142.122\",\"192.168.142.123\",\"192.168.142.124\",\"192.168.142.125\",\"192.168.142.126\",\"192.168.142.127\",\"192.168.142.128\",\"192.168.142.129\",\"192.168.142.130\",\"192.168.142.131\",\"192.168.142.132\",\"192.168.142.133\",\"192.168.142.134\",\"192.168.142.135\",\"192.168.142.136\",\"192.168.142.137\",\"192.168.142.138\",\"192.168.142.139\",\"192.168.142.140\",\"192.168.142.141\",\"192.168.142.142\",\"192.168.142.143\",\"192.168.142.144\",\"192.168.142.145\",\"192.168.142.146\",\"192.168.142.147\",\"192.168.142.148\",\"192.168.142.149\",\"192.168.142.150\",\"192.168.142.151\",\"192.168.142.152\",\"192.168.142.153\",\"192.168.142.154\",\"192.168.142.155\",\"192.168.142.156\",\"192.168.142.157\",\"192.168.142.158\",\"192.168.142.159\",\"192.168.142.160\",\"192.168.142.161\",\"192.168.142.162\",\"192.168.142.163\",\"192.168.142.164\",\"192.168.142.165\",\"192.168.142.166\",\"192.168.142.167\",\"192.168.142.168\",\"192.168.142.169\",\"192.168.142.170\",\"192.168.142.171\",\"192.168.142.172\",\"192.168.142.173\",\"192.168.142.174\",\"192.168.142.175\",\"192.168.142.176\",\"192.168.142.177\",\"192.168.142.178\",\"192.168.142.179\",\"192.168.142.180\",\"192.168.142.181\",\"192.168.142.182\",\"192.168.142.183\",\"192.168.142.184\",\"192.168.142.185\",\"192.168.142.186\",\"192.168.142.187\",\"192.168.142.188\",\"192.168.142.189\",\"192.168.142.190\",\"192.168.142.191\",\"192.168.142.192\",\"192.168.142.193\",\"192.168.142.194\",\"192.168.142.195\",\"192.168.142.196\",\"192.168.142.197\",\"192.168.142.198\",\"192.168.142.199\",\"192.168.142.200\",\"192.168.142.201\",\"192.168.142.202\",\"192.168.142.203\",\"192.168.142.204\",\"192.168.142.205\",\"192.168.142.206\",\"192.168.142.207\",\"192.168.142.208\",\"192.168.142.209\",\"192.168.142.210\",\"192.168.142.211\",\"192.168.142.212\",\"192.168.142.213\",\"192.168.142.214\",\"192.168.142.215\",\"192.168.142.216\",\"192.168.142.217\",\"192.168.142.218\",\"192.168.142.219\",\"192.168.142.220\",\"192.168.142.221\",\"192.168.142.222\",\"192.168.142.223\",\"192.168.142.224\",\"192.168.142.225\",\"192.168.142.226\",\"192.168.142.227\",\"192.168.142.228\",\"192.168.142.229\",\"192.168.142.230\",\"192.168.142.231\",\"192.168.142.232\",\"192.168.142.233\",\"192.168.142.234\",\"192.168.142.235\",\"192.168.142.236\",\"192.168.142.237\",\"192.168.142.238\",\"192.168.142.239\",\"192.168.142.240\",\"192.168.142.241\",\"192.168.142.242\",\"192.168.142.243\",\"192.168.142.244\",\"192.168.142.245\",\"192.168.142.246\",\"192.168.142.247\",\"192.168.142.248\",\"192.168.142.249\",\"192.168.142.250\",\"192.168.142.251\",\"192.168.142.252\",\"192.168.142.253\",\"192.168.142.254\",\"192.168.142.255\"],\"network\":\"192.168.142.0/24\"}]> stderr <>" func="github.com/openshift/assisted-service/internal/bminventory.

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -3043,6 +3042,39 @@ func handleReplyByType(params installer.PostStepReplyParams, b *bareMetalInvento
 	return err
 }
 
+func logReplyReceived(params installer.PostStepReplyParams, log *logrus.Entry) {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("Received step reply <%s> from cluster <%s> host <%s> exit-code <%d>",
```
Comment (ori-amizur) [inline internal/bminventory/inventory.go:3047]: Having the log in a separate function is good, but the usage of the string builder makes it more complicated than it should be.  I think it is better that the output is always written (even if it is empty), except for free-addresses at info level.

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2805,9 +2805,8 @@ func shouldHandle(params installer.PostStepReplyParams) bool {
 }
 
 func (b *bareMetalInventory) PostStepReply(ctx context.Context, params installer.PostStepReplyParams) middleware.Responder {
-	log := logutil.FromContext(ctx, b.log)
-	msg := fmt.Sprintf("Received step reply <%s> from cluster <%s> host <%s>  exit-code <%d> stdout <%s> stderr <%s>", params.Reply.StepID, params.ClusterID,
-		params.HostID, params.Reply.ExitCode, params.Reply.Output, params.Reply.Error)
+	log := logutil.EntryFromContext(ctx, b.log)
```
Comment (ori-amizur) [inline internal/bminventory/inventory.go:2808]: Why not use logutil.FromContext like all other places?

Comment (michaellevy101) [reply internal/bminventory/inventory.go:2808]: FromContext does not support testing agains the actual log level

Comment (ori-amizur) [reply internal/bminventory/inventory.go:2808]: I think there is no problem to write both in debug messages and info messages.

Comment (michaellevy101) [reply internal/bminventory/inventory.go:2808]: sure, but the message will be logged twice

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:3047]: right,  The condition not intended for removing an empty fields but for the log level and the free-addresses special behavior
Removing an empty data is a tiny improvement (We shouldn't write empty data)

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (ori-amizur) [reply internal/bminventory/inventory.go:2808]: It is debug.  We can allow it

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (ori-amizur) [reply internal/bminventory/inventory.go:3047]: I think indication of empty data gives information.  Should not be removed

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:2808]: log.Debugf("%s stdout <%s> ", sb.String(), params.Reply.Output)
log.Info(sb.String())
is this code better?

**Pull request merged**